src/services/url.ts

```diff
@@ -55,6 +55,28 @@ class UrlService extends BaseService<UrlService.Options> {
     }
   }
 
+  handleUrlWithoutListeners() {
+    try {
+      const retVal = <any>this.beautifier.parse<UrlBeautifier.SearchUrlState>(WINDOW().location.href);
+      if ( typeof retVal.then === 'function' ) {
```

Comment:
Unless there are cases where we can't have this be asynchronous, it would be simpler to wrap `retVal` in a Promise using [`Promise.resolve`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve). If `retVal` an object, it will be wrapped in a Promise that resolves to that value; if it's a Promise, then nothing changes and will be returned as-is.

```js
Promise.resolve(retVal).then((resp) => {
  // ...
})
.catch((e) => {
  // ...
});
```